src/mscorlib/shared/System/Diagnostics/StackFrame.cs

```diff
@@ -0,0 +1,272 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+using System.Text;
+using System;
+using System.IO;
+using System.Reflection;
+using System.Diagnostics.Contracts;
+
+namespace System.Diagnostics
+{
+    /// <summary>
+    /// There is no good reason for the methods of this class to be virtual.
+    /// </summary>
+    public partial class StackFrame
+    {
+        private MethodBase method;
+        private int offset;
+        private int ILOffset;
+        private String strFileName;
+        private int iLineNumber;
+        private int iColumnNumber;
+
+        [System.Runtime.Serialization.OptionalField]
+        private bool fIsLastFrameFromForeignExceptionStackTrace;
+
+        internal void InitMembers()
```

Comment:
Does anybody actually call InitMembers() from outside StackTrace? Why can't the constructors that call this chain to the "bool DummyFlag1, bool DummyFlag2" constructor and have it replace "InitMembers" - then maybe some of those fields could be declared "readonly".